examples/lock-app/linux/src/LockEndpoint.cpp

```diff
@@ -457,8 +458,11 @@ bool LockEndpoint::setLockState(DlLockState lockState, const Optional<chip::Byte
         "Lock App: specified PIN code was found in the database, setting door lock state to \"%s\" [endpointId=%d,userIndex=%u]",
         lockStateToString(lockState), mEndpointId, userIndex);
 
-    mLockState = lockState;
-    DoorLockServer::Instance().SetLockState(mEndpointId, mLockState);
+    mLockState                         = lockState;
+    LockOpCredentials userCredential[] = { { CredentialTypeEnum::kPin, uint16_t(credentialIndex) } };
+    auto userCredentials               = MakeNullable<List<const LockOpCredentials>>(userCredential);
+    DoorLockServer::Instance().SetLockState(mEndpointId, mLockState, OperationSourceEnum::kUnspecified,
```

Comment:
Calling SetLockState with this signature will have a different effect than the original code. The original code called SetLockState(chip::EndpointId endpointId, DlLockState newLockState) which *only* sets the attribute and doesn't generate any events. The proposed code here will generate a LockOperation event, which will result in the emission of duplicate events when the lock is operated via Matter (not locally). The signature used here will emit an event if the source!=kRemote. So, ideally this code needs to change to specify the source, in which case the userIndex and credential are only used when the source is local.

Reply:
Good Catch! Validated that there was two events and now only a single one. Updated, can you please review again?